packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -204,22 +254,41 @@ export const BaseInput = ({
         >
           {label}
         </FormLabel>
-        <StyledBaseInput
-          id={inputId}
-          name={name}
-          type={type}
-          defaultValue={defaultValue}
-          value={value}
-          placeholder={placeholder}
+        <BaseInputWrapper
           isDisabled={isDisabled}
           validationState={validationState}
-          isRequired={isRequired}
-          handleOnChange={handleOnChange}
-        />
+          isFocused={isFocused}
+        >
+          <BaseInputVisuals leadingIcon={leadingIcon} prefix={prefix} isDisabled={isDisabled} />
+          <StyledBaseInput
+            id={inputId}
+            name={name}
+            type={type}
+            defaultValue={defaultValue}
+            value={value}
+            placeholder={placeholder}
+            isDisabled={isDisabled}
+            validationState={validationState}
+            isRequired={isRequired}
+            handleOnChange={handleOnChange}
+            leadingIcon={leadingIcon}
+            prefix={prefix}
+            interactionElement={interactionElement}
```

Comment:
So the interaction element will be the `IconButton` component? 

Let's show an example, since IconButton is merged.

Reply:
not necessarily icon button this is baseinput consumer can pass in anything - iconbutton, spinner etc.